### PR TITLE
Failing test case for NodeNameResolver

### PR DESCRIPTION
**Pull request by lulco**
NodeNameResolver's method getName() returns null for MethodCall when its name is Identifier

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Thanks! Could you find the fix too?

**Comment by lulco** [comment]
I'm not sure if it is bug or feature :D There is this condition:

```php
if ($node instanceof MethodCall || $node instanceof StaticCall) {
    if ($this->isCallOrIdentifier($node->name)) {
        return null;
    }
    
    $this->invalidNameNodeReporter->reportInvalidNodeForName($node);
}
```
You probably have some reason to do it, I just don't understand why it is not reporting invalid node for this case or return the correct name

**Comment by TomasVotruba** [comment]
> You probably have some reason to do it, I just don't understand why 

I always forget :), so I use git blame on the line in Github to find the pull-request with "why". My guess that `Identifier` and `Name` are different and should not be mixed in php-parser. They might create different result, e.g. FQN `\string` by default vs. `string`

File: packages-tests/NodeNameResolver/NodeNameResolverTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public function testGetNameOnMethodCallWithIdentifier(): void
+    {
+        $methodCallWithNameIdentifier = new MethodCall(
+            new Variable('foo'),
+            new Identifier('bar')
+        );
+        $this->assertSame('bar', $this->nodeNameResolver->getName($methodCallWithNameIdentifier->name));
+        $this->assertSame('bar', $this->nodeNameResolver->getName($methodCallWithNameIdentifier));
```
Comment (TomasVotruba) [inline packages-tests/NodeNameResolver/NodeNameResolverTest.php]: This line should throw exception, as method call is not a nameable node.

E.g. property fetch is not nameable node either:

```php
$this->value
```

What would be a name? `this`? `value` or `this->value`?

There was lot of bugs with assumption, that MethodCall returns left or right part, so this was added to make sure the `name` is needed, nod the variable part. Mainly in chained calls this caused mess:

```php
$this->createBuilder()
    ->select('...')
    ->from(some()->value())`
```

So we better go narrow than open space for bugs :)

File: packages-tests/NodeNameResolver/NodeNameResolverTest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        $this->nodeNameResolver = $this->getService(NodeNameResolver::class);
+    }
+
+    public function testGetNameOnMethodCallWithIdentifier(): void
+    {
+        $methodCallWithNameIdentifier = new MethodCall(
+            new Variable('foo'),
+            new Identifier('bar')
+        );
+        $this->assertSame('bar', $this->nodeNameResolver->getName($methodCallWithNameIdentifier->name));
```
Comment (TomasVotruba) [inline packages-tests/NodeNameResolver/NodeNameResolverTest.php]: This should work :+1:

**Comment by TomasVotruba** [comment]
Just a ping in case you've missed it. This PR fails correctly  :+1: , so it should be changed or closed

**Comment by lulco** [comment]
Imho it should throw an exception, I can rewrite the test later

**Comment by TomasVotruba** [comment]
I think we throw exceptions before, but it was unclear to people what they should put in.
Check the git blame on this services, there are some PRs related.

File: packages-tests/NodeNameResolver/NodeNameResolverTest.php (hunk 1 above)
Comment (lulco) [reply packages-tests/NodeNameResolver/NodeNameResolverTest.php]: Exactly, it should throw exception :)

**Comment by lulco** [comment]
> I think we throw exceptions before, but it was unclear to people what they should put in.
> Check the git blame on this services, there are some PRs related.

For me it is unclear when I get null :) exception can have some message which can be improved, but null is just null :)
I've rewrite test to expect exception. Didn't check PRs for now.

**Comment by TomasVotruba** [comment]
I'm going through old PRs to narrow focus on what can be merged. 
Closing as 2 month later and unfinished. 
Feel free to re-open, just please make the CI passing, so we can solve it right away :+1:

**Pull request closed without merging**